client.go

```diff
@@ -1830,3 +1836,41 @@ func flags(f int) uint32 {
 	}
 	return out
 }
+
+// unixPerm converts Go permission bits to Unix permission bits,
+// masking off file type bits.
+//
+// The SFTP protocol draft doesn't specify the exact values of the bits.
+// POSIX is fuzzy in this regard: it specifies bits for the chmod command [1],
+// but not for C applications [2]. We follow the chmod bits, which are the
+// bits that OpenSSH's sftp-server understands when running on Unix, in the
+// hopes that other implementations simulate its behavior.
+//
+// This differs from fromFileMode in that it preserves the POSIX versions of
+// setuid, setgid and sticky in m, because we've historically supported those
+// bits.
+//
+// [1] https://pubs.opengroup.org/onlinepubs/009604499/utilities/chmod.html
+// [2] https://pubs.opengroup.org/onlinepubs/009604499/basedefs/sys/stat.h.html
+func unixPerm(m os.FileMode) uint32 {
```

Comment:
Calling them POSIX bits is misleading, though. The actual bit values aren't really specified by POSIX, as the comment above explains. They're more of a Unix tradition.

Reply:
Rest assured, they did get codified to the same values later: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_stat.h.html

Reply:
:facepalm: Always check the right version of the spec.

Reply:
I mean, the SFTP doc absolutely references the versions of the specs that you link to here, but POSIX went ahead and said, “everyone is using these values, because it just makes sense to use the exact same values as the `chmod` command-line tool.” So, they went ahead and just turned what everyone was already doing into formal codification.

I really only know about this because of the `S_IFMT` shenanigans back a bit ago. 😆